lib/msf/core/module_set.rb

```diff
@@ -34,16 +34,12 @@ def [](name)
   # @param reference_name [String] The module reference name.
   # @return [Msf::Module,nil] Instance of the named module or nil if it
   #   could not be created.
-  def create(reference_name)
+  def create(reference_name, use_cache: true)
     klass = fetch(reference_name, nil)
     instance = nil
-
-    # If there is no module associated with this class, then try to demand
-    # load it.
+    # If there is no module associated with this class, then try to demand load it.
     if klass.nil? or klass == Msf::SymbolicModule
-      if framework.modules.load_cached_module(module_type, reference_name) || empty?
-        recalculate
```

Comment:
Just to confirm; do we not want this behavior around anymore? 🤔 

The theory edgecase is; someone adds additional modules to the loadpath out of band, i.e. via a plugin or something, and we need to ensure we call recalculate still

Reply:
Thought about this a bit more and I reckon it does make sense to `recalculate` here if the payload can't be loaded from the cache, this is particularly useful in the case when someone adds a new stager to their `~/.msf4/modules` directory the first time they try to use any payload with that stager all combinations will be recalculated and stored in the cache (if the async cache refresh didn't get to it first which it likely would anyway)

Reply:
Ignore my previous comment, adding the recalculate back in resulted in a regression where unknown commands took several secodns to return due to calling `recalculate` so I've removed it again